src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -799,28 +800,68 @@ export default class Fms {
      * @param routeString {string}
      * @return {boolean}
      */
-    replaceDepartureProcedure(routeString) {
-        const routeStringElements = routeString.split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
+    replaceDepartureProcedure(routeString, airportIcao) {
+        const routeStringElements = routeString.toUpperCase().split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
 
-        if (routeStringElements.length !== 3) {
+        if (routeStringElements.length > 3) {
             return [false, 'departure procedure format not understood'];
         }
 
-        const procedureId = routeStringElements[1];
+        let procedureId = routeStringElements[0];
 
-        if (!NavigationLibrary.hasProcedure(procedureId)) {
+        if (routeStringElements.length === 3) {
+            procedureId = routeStringElements[1];
+        }
+
+        const sidModel = NavigationLibrary.getProcedure(procedureId);
+
+        if (_isNil(sidModel)) {
             return [false, `unknown procedure "${procedureId}"`];
         }
 
-        const wasSuccessful = this._routeModel.replaceDepartureProcedure(routeString);
+        if (routeStringElements.length === 1) { // RouteString looks like PROC
+            const exitPoint = _findLast(this.waypoints, (waypointModel) => sidModel.hasExit(waypointModel.name));
 
-        if (wasSuccessful) {
-            this._updateDepartureRunwayFromRoute();
+            if (!exitPoint) {
+                return [false, `the ${procedureId.toUpperCase()} departure doesn't have an exit along our route`];
+            }
 
-            return [true, ''];
+            routeStringElements.push(exitPoint.name);
         }
 
-        return [false, `route of "${routeString}" is not valid`];
+        if (routeStringElements.length === 2) { // RouteString looks like PROC.EXIT
+            const runwayModel = this.departureRunwayModel;
+
+            if (_isNil(runwayModel)) {
+                return [false, 'unsure if we can accept that procedure; we don\'t have a runway assignment'];
+            }
+
+            let entryName = sidModel.getUniqueEntryPoint();
+
+            if (_isEmpty(entryName)) {
+                entryName = airportIcao.toUpperCase() + runwayModel.name;
```

Comment:
L839-L852 don't make sense to me. If they're assigned a new route on the ground, they should accept it and expect to be taxiied to a runway which is valid for that SID. Thus, they should not reject the route as they do below. The behavior as written here is this:
- If there is only one runway which is valid for the requested SID, then expect that runway and enter the route into the FMS
- If there either multiple runways, or _no_ runways, which are valid for the requested SID, reject the route.

It should be this:
- If the current expected departure runway _is_ valid for the requested SID, then apply the route using the currently expected departure runway.
- If the current expected departure runway is not valid for the requested SID, expect any runway which _is_ valid. The aircraft must not be allowed to depart until it is taxiied to a runway which is valid for the SID.